java/core/src/java/org/apache/orc/OrcFile.java

```
@@ -642,4 +671,187 @@ public static Writer createWriter(Path path,
+    Map<String, ByteBuffer> userMetadata = new HashMap<>();
+
+    for (Path input: inputFiles) {
+      FileSystem fs = input.getFileSystem(conf);
+      Reader reader = createReader(input,
+          readerOptions(options.getConfiguration()).filesystem(fs));
+
+      if (!understandFormat(input, reader)) {
+        continue;
+      } else if (schema == null) {
+        // if this is the first file that we are including, grab the values
+        schema = reader.getSchema();
+        compression = reader.getCompressionKind();
+        bufferSize = reader.getCompressionSize();
+        rowIndexStride = reader.getRowIndexStride();
+        fileVersion = reader.getFileVersion();
+        writerVersion = reader.getWriterVersion();
+        options.blockSize(bufferSize)
+            .version(fileVersion)
+            .writerVersion(writerVersion)
+            .compress(compression)
+            .rowIndexStride(rowIndexStride)
+            .setSchema(schema);
+        if (compression != CompressionKind.NONE) {
+          options.enforceBufferSize().bufferSize(bufferSize);
+        }
+        mergeMetadata(userMetadata, reader);
+        output = createWriter(outputPath, options);
+      } else if (!readerIsCompatible(schema, fileVersion, writerVersion,
+          bufferSize, rowIndexStride, compression, userMetadata, input,
+          reader)) {
+        continue;
+      } else {
+        mergeMetadata(userMetadata, reader);
+      }
+      List<OrcProto.StripeStatistics> statList =
+          reader.getOrcProtoStripeStatistics();
+      FSDataInputStream inputStream = fs.open(input);
+      int stripeNum = 0;
+      result.add(input);
+
+      for(StripeInformation stripe: reader.getStripes()) {
+        int length = (int) stripe.getLength();
+        if (buffer.length < length) {
+          buffer = new byte[length];
+        }
+        long offset = stripe.getOffset();
+        inputStream.readFully(offset, buffer, 0, length);
```

No, I'd rather have the whole method blow out than hide the exception. It can only happen in the case of a broken ORC file.